src/inet/networklayer/generic/GenericRoutingTable.cc

```diff
@@ -145,13 +145,15 @@ void GenericRoutingTable::configureRouterId()
     if (routerId.isUnspecified()) {    // not yet configured
         const char *routerIdStr = par("routerId").stringValue();
         if (!strcmp(routerIdStr, "auto")) {    // non-"auto" cases already handled in earlier stage
-            // choose highest interface address as routerId
+            // choose lowest interface address as routerId
             for (int i = 0; i < ift->getNumInterfaces(); ++i) {
                 InterfaceEntry *ie = ift->getInterface(i);
                 if (!ie->isLoopback()) {
                     L3Address interfaceAddr = ie->getGenericNetworkProtocolData()->getAddress();
-                    if (routerId.isUnspecified() || routerId < interfaceAddr)
+                    if (routerId.isUnspecified() || routerId < interfaceAddr) {
                         routerId = interfaceAddr;
+                        break;
```

Comment:
I think, this "break" is not a good idea here.
Function returns first non-loopback address instead of the lowest address.